burn-book/src/import/onnx-model.md

```diff
@@ -0,0 +1,135 @@
+is precisely the scenario where importing models into Burn can be a game-changer.
+
+## Traditional Methods: The Drawbacks
+
+If you've been working with other deep learning frameworks like PyTorch, it's likely that you've
+exported model weights before. PyTorch, for instance, lets you save model weights using its
+`torch.save()` function. Yet, to port this model to another framework, you face the arduous task of
+manually recreating the architecture in the destination framework before loading in the weights. Not
+only is this method tedious, but it's also error-prone and hinders smooth interoperability between
+frameworks.
+
+## Enter ONNX: The One Format to Rule Them All
+
+ONNX (Open Neural Network Exchange) is designed to solve such complications. It's an open-standard
+format that exports both the architecture and the weights of a deep learning model. This feature
+makes it exponentially easier to move models between different frameworks, thereby significantly
+aiding interoperability. ONNX is supported by a number of frameworks including but not limited to
+TensorFlow, PyTorch, Caffe2, and Microsoft Cognitive Toolkit.
+
+### Advantages of ONNX
+
+ONNX stands out for encapsulating two key elements:
+
+1. **Model Information**: It captures the architecture, detailing the layers, their connections, and
+   configurations.
+2. **Weights**: ONNX also contains the trained model's weights.
+
+This dual encapsulation not only simplifies the porting of models between frameworks but also allows
+seamless deployment across different environments without compatibility concerns.
+
+## Burn's ONNX Support: Importing Made Easy
+
+Understanding the important role that ONNX plays in the contemporary deep learning landscape, Burn
+simplifies the process of importing ONNX models via an intuitive API designed to mesh well with
+Burn's ecosystem.
+
+Unique to Burn is its approach of converting an ONNX file into Rust source code alongside
+Burn-compatible weights. This conversion is facilitated by the `burn-import` crate's code generator,
+executed during build time.
+
+### Advantages of Burn's ONNX Approach
+
+1. **Native Integration**: The generated Rust code is fully integrated into Burn's architecture,
+   enabling your model to run on various backends without the need for a separate ONNX runtime.
+
+2. **Trainability**: The imported model is not just for inference; it can be further trained or
+   fine-tuned using Burn's native training loop.
+
+3. **Portability**: As the model is converted to Rust source code, it can be compiled into
+   WebAssembly for browser execution. Likewise, this approach is beneficial for no-std embedded
+   devices.
+
+4. **Optimization**: Rust's compiler can further optimize the generated code for target
+   architectures, thereby improving performance.
+
+As far as we know, Burn is the only deep learning framework that employs this technique, making it
```

Comment:
Unclear which technique you are referring to

Reply:
Yeah, I think a sentence is necessary at the beginning of the paragraph on code generation.